### PR TITLE
[MIG] base_ubl: Migration to 15.0

### PR DESCRIPTION
**Pull request by tbanevicius**
Migration of module base_ubl to 15.0 Odoo version

### CODE REVIEW

**Comment by tbanevicius** [comment]
Hi @sbidoul @pedrobaeza , maybe you know why no checks are running on this Merge Request ?

**Comment by tbanevicius** [comment]
Hi @phuctranfxvn @astirpe @alexis-via . Could someone review this merge request?

**Comment by tbanevicius** [comment]
Hi @jbaudoux . Could you review it?

**Review by jbaudoux — APPROVED** [review]
LGTM

**Comment by tbanevicius** [comment]
Hi @pedrobaeza can you merge this MR?

**Comment by pedrobaeza** [comment]
It requires 2 approvals, one of them by a PSC.

**Comment by tbanevicius** [comment]
Hi @gurneyalex . Could you review this issue? thanks.

**Comment by simahawk** [comment]
I'm having a look. Could you please squash the fixup commits? Thanks :pray:

File: base_ubl/models/ubl.py [hunk 1]
```diff
@@ -260,6 +260,8 @@ def _ubl_add_supplier_party(
                 ), "partner is wrong"
             else:
                 partner = company.partner_id
+        else:
```
Comment (simahawk) [inline base_ubl/models/ubl.py:263]: when does it happen?

File: base_ubl/models/ubl.py [hunk 2]
```diff
@@ -823,6 +825,6 @@ def get_xml_files_from_pdf(self, pdf_file):
                 except Exception:
                     continue
         except Exception:
-            pass
+            logger.debug("Passing any uncaught exception.")
```
Comment (simahawk) [inline base_ubl/models/ubl.py]: this logging is kind of useless: could you log the error?

**Review by simahawk — CHANGES_REQUESTED** [review]
LG, minor remarks

**Comment by simahawk** [comment]
> I'm having a look. Could you please squash the fixup commits? Thanks pray

bah, never mind. You are not adding those commits and is going to make our life w/ `oca-port` tool more complicated (it won't detect equal commits from version to version)

File: base_ubl/models/ubl.py (hunk 1 above)
Comment (tbanevicius) [reply base_ubl/models/ubl.py:263]: Basically if company is not passed to method  _ubl_add_supplier_party error occured. So I've added custom error for making company field required for this method.

**Comment by tbanevicius** [comment]
Hi @simahawk . By any chance this can be merged?

**Review by simahawk — APPROVED** [review]

**Comment by espo-tony** [comment]
Someone can say why the merge of this failed?

**Comment by simahawk** [comment]
Quite an old build, let's try again.

/ocabot merge nobump

**Comment by simahawk** [comment]
https://github.com/OCA/edi/runs/7859320796?check_suite_focus=true#step:8:233

```
2022-08-16 13:47:03,429 282 ERROR odoo unittest.suite: ERROR: setUpClass (odoo.addons.base_ubl.tests.test_ubl_generate.TestUblInvoice)
Traceback (most recent call last):
  File "/__w/edi/edi/base_ubl/tests/test_ubl_generate.py", line 21, in setUpClass
    super().setUpClass(chart_template_ref=chart_template_ref)
  File "/opt/odoo/addons/account/tests/common.py", line 31, in setUpClass
    assert 'post_install' in cls.test_tags, 'This test requires a CoA to be installed, it should be tagged "post_install"'
AssertionError: This test requires a CoA to be installed, it should be tagged "post_install"
```

**Comment by aisopuro** [comment]
@simahawk @dreispt I opened a new PR with fixes to the tests, since I don't have access to push to this branch: #730 

I've kept the original authorship intact to avoid taking undue credit.

**Comment by dreispt** [comment]
Superseded by #730

**Pull request closed without merging**